### ISSUE TITLE
make it installable with pip or at least link download page

### ISSUE THREAD

**Issue by matkoniecz** [nltk/nltk#2228]
To reproduce:

- start with an environment without `nltk`
- run `sudo pip install nltk`
- create file with simple `nltk` script (see below)
- run this script

I tested with 

```import nltk

print(nltk.word_tokenize("test string     aaaaa\n\n\nfinal."))
``` 

based on https://stackoverflow.com/a/37559340/4130619

I got error where relevant part is 
```
LookupError: 
**********************************************************************
  Resource punkt not found.
  Please use the NLTK Downloader to obtain the resource:

  >>> import nltk
  >>> nltk.download('punkt')
  
  Attempted to load tokenizers/punkt/PY3/english.pickle
```

It is surprising as I would expect pip to install package and all necessary resources, not only parts of it.

If it is really necessary to make installation more complicated - please link the real download page or help page in the error message.

full error message: https://pastebin.com/raw/VsjrxvZ5

**Comment by purificant** [comment]
Some dependencies include data sets that are several gigabytes in size, it makes sense to not install them every time the main package is installed and allow granular downloads depending on use case. The error message shows which data resource is missing and how it can be downloaded. 

The installation docs are here: http://www.nltk.org/install.html
The docs about data downloads are one below in the menu on the right* hand side: http://www.nltk.org/data.html
For more technically minded users, the instructions on setting up the dev environment also include downloading the datasets: https://github.com/nltk/nltk/blob/develop/CONTRIBUTING.md

Whilst I appreciate the initial surprise at seeing the error message, the message itself does show how to fix it.

**Comment by matkoniecz** [comment]
tl;dr: 
I think that https://www.nltk.org/data.html should be linked in this error message to make it less surprising.

-------

http://www.nltk.org/install.html has just 

```

    Install NLTK: run sudo pip install -U nltk
    Install Numpy (optional): run sudo pip install -U numpy
    Test installation: run python then type import nltk

For older versions of Python it might be necessary to install setuptools (see http://pypi.python.org/pypi/setuptools) and to install pip (sudo easy_install pip).
```

so it is not too useful and "NLTK Downloader" is not even mentioned there.

"NLTK Downloader" is also not mentioned at https://github.com/nltk/nltk/wiki/Installing-Third-Party-Software linked from that page.

I think that https://www.nltk.org/data.html should be linked in this error message.

**Comment by purificant** [comment]
> tl;dr:
> I think that https://www.nltk.org/data.html should be linked in this error message to make it less surprising.

I think this makes sense to add. Proposed patch here: https://github.com/nltk/nltk/pull/2230
Suggestions on the message text are welcome.

**Comment by alvations** [comment]
Closing issue as resolved, since #2230 is merged. 

Please re-open if the issue is still relevant/unresolved.

**Comment by MSch** [comment]
Just wanted to restart discussion / reopen this, because I ran into this precise problem just now.

Having a built-in downloader makes it harder than IMHO necessary to depend on nltk. I understand that some dependencies are several gigabytes in size, but even that should be manageable with a package manager like poetry/nix reusing a local download.

My vote would be to optionally allow depending on individual nltk data as packages, so that I can e.g. depend on `nltk-punkt-english` and know that a certain version of this data has been "locked" in my poetry lockfile.

**Comment by logan-markewich** [comment]
I agree with @MSch 

Having something like `pip install nltk[punkt]` would be extremely valuable
